### ISSUE TITLE
response too large (at least 1020631 bytes) when adding address 1Q7zx6cUVXDZpAJHgPQwbbZUfTKFFuEV6z

### ISSUE THREAD

**Issue by circulosmeos** [spesmilo/electrum#4315]
This error is shown when running "electrum -v" **v3.1.3** (Windows or linux)
after adding address *1Q7zx6cUVXDZpAJHgPQwbbZUfTKFFuEV6z*:

`[Synchronizer] response error: {'error': {'code': -32600, 'message': 'response too large (at least 1020631 bytes)'}, 'method': 'blockchain.scripthash.get_history', 'jsonrpc': '2.0', 'params': ['1Q7zx6cUVXDZpAJHgPQwbbZUfTKFFuEV6z'], 'id': 39}`

**Note that this address has 39374 transactions.**

The balance shown is zero (and no transactions in History are shown):
[![enter image description here][2]][2]
but an invalid balance is shown when adding more addresses: for example after adding *1GK3btZr57VDEQvwpib6Qhj479Ckzt8z2n* the balance is 10 bits, when it should be 0:

[![enter image description here][4]][4]
[![enter image description here][5]][5]

  [2]: https://i.stack.imgur.com/SYJIl.png
  [4]: https://i.stack.imgur.com/4SM7S.png
  [5]: https://i.stack.imgur.com/ieluV.png

**Comment by SomberNight** [comment]
Yes, the current protocol is not suitable for addresses with long histories unfortunately.
Related https://github.com/kyuupichan/electrumx/issues/348

**Comment by SomberNight** [comment]
Note that there is a max message size restriction enforced on BOTH client-side and server-side, by default of ~1 MB.

-----

client-side: https://github.com/spesmilo/electrum/blob/72950bf3794b2a2dce67c152ebc8fadb8d31b8a1/electrum/interface.py#L70

-----

server-side: https://electrumx-spesmilo.readthedocs.io/en/latest/environment.html#envvar-MAX_SEND

**Comment by SomberNight** [comment]
in case the server raises, the log in current code looks like this:
`E/i | interface.[fortress.qtornado.com:443] | disconnecting due to GracefulDisconnect(RPCError(1, 'history too large'))`

in case the server uses higher limits than the client, and it is only the client that raises, the exception is:
`MemoryError('dropping message over 1,000,000 bytes and re-synchronizing',)`

**Comment by johnzweng** [comment]
Hi @SomberNight ! 🙂 

As I see that the limit on electrumX can be configured via `ENV` variable, would it be possible to also allow the limit on the client side to be configurable (either via `ENV` variable or `config` file)?

This would allow powerusers which have to work with addresses with large histories to tune their own setup (server + client) to work around this problem without needing to maintain their own electrum branch (where the limit is increased).

**Note**: _I see this problem mostly as a legacy problem as thanks to HD wallets we hopefully won't see this large histories anymore in future, but for some non-technical reasons, we still need to be able to monitor old legacy addresses (which unfortunately have already large histories)._

**Comment by SomberNight** [comment]
@johnzweng done in https://github.com/spesmilo/electrum/commit/5a7c3dc4d0bcd0ab3f60108ab99bbf783ba9855c

**Comment by johnzweng** [comment]
This was quick! Thanks a lot! 🙂 👍

**Comment by SomberNight** [comment]
note: electrum protocol 1.5 would resolve this
https://github.com/spesmilo/electrumx/pull/90
